### PR TITLE
Create cf-1462E2

### PR DESCRIPTION
**Pull request by EulerPascal404**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Review by dongliu0426 — CHANGES_REQUESTED** [review]
also please rename the file to `solutions/cf-1462E2.mdx`

**Comment by EulerPascal404** [comment]
> also please rename the file to `solutions/cf-1462E2.mdx`

yikes, completely forgot. Fixed that now.

**Review by dongliu0426 — APPROVED** [review]

File: solutions/cf-1462E2.mdx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		freq[nums[i]]++;
+	}
+	vector<int> preFreq(n + 1, 0);  // prefix-array for freq vector;
+	for(int i = 1; i <= n; i++){
+		preFreq[i] = freq[i] + preFreq[i-1];
+	}
+	for(int i = 1; i <= n; i++){
+		int max_index = min(n, i + k);  // the max possible value that could be in a tuple with i as the minimum
+		int num_total_elems = preFreq[max_index] - preFreq[i-1];  // the number of elements from [i, max_index]
+		int num_larger_elems = preFreq[max_index] - preFreq[i];  // the number of elements from [i+1, max_index]
```
Comment (ryanchou-dev) [inline solutions/cf-1462E2.mdx:77]: over 80 chars

File: solutions/cf-1462E2.mdx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	}
+	cout << ans << "\n";
+}
+
+int main() {
+	factorial();
+	inverses();
+	int t;
+	cin >> t;
+	while(t--){
```
Comment (ryanchou-dev) [inline solutions/cf-1462E2.mdx:96]:
```suggestion
	
	while(t--) {
```

**Review by ryanchou-dev — APPROVED** [review]

File: solutions/cf-1462E2.mdx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+**Time Complexity:** $\mathcal{O}(N\log p)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
```
Comment (Aadit-Ambadkar) [inline solutions/cf-1462E2.mdx:20]:
```suggestion
typedef long long ll;
```
depending on the ide, this does make a difference. But functionally no dif.

**Comment by EulerPascal404** [comment]
I think this can be merged now @dongliu0426

File: solutions/cf-1462E2.mdx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	cin >> n >> m >> k;
+	vector<int> nums(n);
+	vector<int> freq(n + 1, 0);  // frequency of each element
+	for(int i = 0; i < n; i++) {
+		cin >> nums[i];
+		freq[nums[i]]++;
+	}
+	vector<int> preFreq(n + 1, 0);  // prefix-array for freq vector;
+	for(int i = 1; i <= n; i++){
+		preFreq[i] = freq[i] + preFreq[i-1];
```
Comment (dongliu0426) [inline solutions/cf-1462E2.mdx:72]:
```suggestion
		preFreq[i] = freq[i] + preFreq[i - 1];
```

File: solutions/cf-1462E2.mdx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+	}
+	for(int i = 1; i <= n; i++){
+		int max_index = min(n, i + k);  // the max possible value that could be in a tuple with i as the minimum
+		int num_total_elems = preFreq[max_index] - preFreq[i-1];  // the number of elements from [i, max_index]
```
Comment (dongliu0426) [inline solutions/cf-1462E2.mdx:76]:
```suggestion
		int num_total_elems = preFreq[max_index] - preFreq[i - 1];  // the number of elements from [i, max_index]
```

File: solutions/cf-1462E2.mdx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+	return fac[n] * inv[r] % MOD * inv[n - r] % MOD;
+}
+
+void solve(){
+	int ans = 0;
+	int n, m, k;
+	cin >> n >> m >> k;
+	vector<int> nums(n);
+	vector<int> freq(n + 1, 0);  // frequency of each element
+	for(int i = 0; i < n; i++) {
```
Comment (dongliu0426) [inline solutions/cf-1462E2.mdx:66]:
```suggestion
	for (int i = 0; i < n; i++) {
```

File: solutions/cf-1462E2.mdx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+		cin >> nums[i];
+		freq[nums[i]]++;
+	}
+	vector<int> preFreq(n + 1, 0);  // prefix-array for freq vector;
+	for(int i = 1; i <= n; i++){
```
Comment (dongliu0426) [inline solutions/cf-1462E2.mdx:71]:
```suggestion
	for (int i = 1; i <= n; i++) {
```

File: solutions/cf-1462E2.mdx [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+		preFreq[i] = freq[i] + preFreq[i-1];
+	}
+	for(int i = 1; i <= n; i++){
```
Comment (dongliu0426) [inline solutions/cf-1462E2.mdx:74]:
```suggestion
	for (int i = 1; i <= n; i++) {
```

**Review by dongliu0426 — APPROVED** [review]

**Comment by ryanchou-dev** [comment]
why the request? i already approved.

**Comment by dongliu0426** [comment]
idk I'll merge after vercel deploys

**Pull request merged**